incremental-font-transfer/src/glyph_keyed.rs

```diff
@@ -264,160 +263,138 @@ fn retained_glyphs_total_size<T: GlyphDataOffsetArray>(
             .checked_sub(start_offset) // TODO: this can be removed if we pre-verify ascending order
             .ok_or(PatchingError::FontParsingFailed(ReadError::MalformedData(
                 "offset entries are not in ascending order",
-            )))? as u64;
+            )))? as usize;
     }
 
     Ok(total_size)
 }
 
-/// Objects with this trait can be written to bytes.
-///
-/// The offset value will NOT be divided prior to conversion to bytes.
-trait WritableOffset {
-    fn write_to(self, dest: &mut [u8]);
+struct OffsetArrayAndData {
+    data: Vec<u8>,
+    offset_array: Vec<u8>,
 }
 
-impl WritableOffset for u32 {
-    fn write_to(self, dest: &mut [u8]) {
-        let data: [u8; 4] = self.to_raw();
-        dest[..4].copy_from_slice(&data);
-    }
-}
-
-impl WritableOffset for Uint24 {
-    fn write_to(self, dest: &mut [u8]) {
-        let data: [u8; 3] = self.to_raw();
-        dest[..3].copy_from_slice(&data);
-    }
+struct OffsetArrayBuilder<'a, T: GlyphDataOffsetArray> {
```

Comment:
unless it's necessary for some reason (often involving associated types) in general if you can omit a bound on a generic in a type definition you should (since it then ends up being required everywhere else you might impl something for a type, including unrelated traits etc)